tests/ui/implied-bounds/implied-bounds-on-trait-hierarchy.rs

```diff
@@ -1,9 +1,7 @@
-// check-pass
```

Comment:
#84591 is still unsound, even with this PR, is it? do we have a test which still incorrectly compiles. It feels like we're not fixing this issue but only making it harder to trigger